File: 프로그래머스/unrated/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.js
function solution(bandage, health, attacks) {
    var cnt = 0;
    // 마지막 공격까지의 시간
    for(var i = 0; i < attacks.length; i++){
        var attack = attacks[i];
        if(cnt < attack[0]){
            cnt = attack[0];    
        }console.log(cnt);     
    }
    
    var health_now = health; // 현재 체력
    var bandage_time = 1; // 붕대감기 시간
    var at_t = 0; // 공격할 시간
    
    // 마지막 공격까지의 1초마다의 반복문
    for(var i = 0; i <= cnt; i++){
        console.log("time: ",i);
        if (i == attacks[at_t][0]){ // 공격할 시간
            health_now -= attacks[at_t][1];
            bandage_time = 1;
            at_t++;
        }else{
            if(bandage_time == bandage[0]){ // 붕대감기 시간이 연속 붕대감
                health_now += (bandage[2] + bandage[1]);
                bandage_time = 1;
            }else{
                health_now += bandage[1];
                bandage_time++;
            }
            if(health_now >= health){
                health_now = health;
            } 
        }
        console.log(health_now);
        if(health_now <= 0){
            return -1;
            break;
        }
    } 
    return health_now;
}

File: 프로그래머스/unrated/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.py
def solution(bandage, health, attacks):
    answer,a,b_t,time = 0,0,1,0
    max_t=attacks[-1][0]
    max_h=health
    while time<=max_t:
        if time==attacks[a][0]:
            health-=attacks[a][1]
            b_t=1
            time+=1
            a+=1
        else:
            if b_t==bandage[0]:
                health+=(bandage[1]+bandage[2])
                b_t=1
            else:
                health+=bandage[1]
            b_t+=1
            time+=1
        if health>=max_h: health=max_h
        if health<=0: return -1  
    return health